ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/widgets/MultiFileSelectorField.java

```diff
@@ -130,23 +133,25 @@ public FileSelectDialogHideHandler(final FileSelectDialog dlg,
         }
 
         @Override
-        public void onHide(HideEvent event) {
-            Set<File> files = diskResourceUtil.filterFiles(dlg.getDiskResources());
-            if (files.isEmpty()) {
-                return;
-            }
-            store.addAll(files);
-            if (userSettings.isRememberLastPath() && store.size() > 0) {
-                userSettings.setLastPath(diskResourceUtil.parseParent(store.get(0).getPath()));
-                eventBus.fireEvent(new LastSelectedPathChangedEvent(true));
-            }
-            ValueChangeEvent.fire(MultiFileSelectorField.this,
-                                  Lists.<HasPath> newArrayList(store.getAll()));
-            setValue(Lists.<HasPath> newArrayList(store.getAll()));
+        public void onDialogHide(DialogHideEvent event) {
+        	if(event.getHideButton().equals(PredefinedButton.OK)) {
+	            Set<File> files = diskResourceUtil.filterFiles(dlg.getDiskResources());
+	            if (files.isEmpty()) {
+	                return;
+	            }
+	            store.addAll(files);
+	            if (userSettings.isRememberLastPath() && store.size() > 0) {
+	                userSettings.setLastPath(diskResourceUtil.parseParent(store.get(0).getPath()));
+	                eventBus.fireEvent(new LastSelectedPathChangedEvent(true));
+	            }
```

Comment:
I just checked the behavior in other places, it does look like we only save last path only when user hits 'OK' button. So I will stick to that.